Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp

```diff
@@ -122,6 +125,32 @@ class MemoryViewTable final : public QTableWidget
     }
   }
 
+  void OnItemChanged(QTableWidgetItem* item)
+  {
+    if (m_view->m_updating == true)
+      return;
+
+    m_view->m_updating = true;
+    QString text = item->text();
+    u32 address = item->data(USER_ROLE_CELL_ADDRESS).toUInt();
+    int type = item->data(USER_ROLE_VALUE_TYPE).toInt();
+    std::vector<u8> bytes;
+
+    bytes = m_view->ConvertTextToBytes(type, text);
+
+    // Cannot be used while update is running.
+    AddressSpace::Accessors* accessors = AddressSpace::GetAccessors(m_view->GetAddressSpace());
+
+    if (accessors->IsValidAddress(address) && !bytes.empty())
+    {
+      for (const u8 c : bytes)
+        accessors->WriteU8(address++, c);
+    }
```

Comment:
You need to re-verify the address for each byte, you could write past the end of memory.

Reply:
Interesting. I accidentally carried the check over from loading memory values. In the current memory widget coding, there's no check at all. I'm not sure how it handles input that starts in range then goes out of range.  

I don't think I need the check at all. You can only edit a cell if it is a valid address, and you can only go past the cell size if you input ascii. but I'm not sure if that should be allowed?

Reply:
Ok so I stepped through it in the VS debugger. It does kind of seem to freak out. I added an end address check, plus added address checks to MemoryWidget.  I don't think we want to write anything if the end address is invalid.